src/sinks/new_relic_logs.rs

```diff
@@ -0,0 +1,97 @@
+use crate::{
+    buffers::Acker,
+    sinks::http::{Encoding, HttpMethod, HttpSinkConfig},
+    sinks::util::{BatchConfig, Compression, TowerRequestConfig},
+    topology::config::{DataType, SinkConfig, SinkDescription},
+};
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+#[derivative(Default)]
+pub enum NewRelicLogsRegion {
+    #[derivative(Default)]
+    Us,
+    Eu,
+}
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone)]
+pub struct NewRelicLogsConfig {
+    pub license_key: Option<String>,
+    pub insert_key: Option<String>,
+    pub region: Option<NewRelicLogsRegion>,
+
+    #[serde(default, flatten)]
+    pub batch: BatchConfig,
+
+    #[serde(flatten)]
+    pub request: TowerRequestConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<NewRelicLogsConfig>("new_relic_logs")
+}
+
+#[typetag::serde(name = "new_relic_logs")]
+impl SinkConfig for NewRelicLogsConfig {
+    fn build(&self, acker: Acker) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let mut headers: IndexMap<String, String> = IndexMap::new();
+
+        if let Some(license_key) = &self.license_key {
+            headers.insert("X-License-Key".to_owned(), license_key.clone());
+        } else if let Some(insert_key) = &self.insert_key {
+            headers.insert("X-Insert-Key".to_owned(), insert_key.clone());
+        } else {
+            return Err(format!("must provide either 'license_key' or 'insert_key'").into());
```

Comment:
Should we use something like Snafu here to have structured errors?

Reply:
Will do